pkg/pillar/agentlog/loglevel.go

```diff
@@ -69,98 +75,101 @@ func getRemoteLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetRemoteLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
+	gc := cast.CastConfigItemValueMap(m)
 	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.RemoteLogLevel != "" {
-		return as.RemoteLogLevel, true
-	}
-	// Do we have a default value?
-	if allowDefault && gc.DefaultRemoteLogLevel != "" {
-		return gc.DefaultRemoteLogLevel, true
+	if _, ok1 := gc.AgentSettings[agentName]; ok1 {
+		as, ok := gc.AgentSettings[agentName]["remote.loglevel"]
+		if ok && as.StringValue != "" {
+			return as.Key, true
+		} else if ok && as.StringValue != "" && allowDefault {
+			// Return default remote.loglevel (global)
+			return "global", true
+		} else {
+			log.Infof("Failed to find loglevel settings for %s", agentName)
+		}
+	} else {
+		log.Infof("Failed to find per agent settings for %s", agentName)
 	}
 	return "", false
 }
 
-func LogLevel(gc *types.GlobalConfig, agentName string) string {
-
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.LogLevel != "" {
-		return as.LogLevel
+func LogLevel(gc *types.ConfigItemValueMap, agentName string) string {
+	as := types.GetAgentValue("debug" + agentName + "loglevel", agentName, gc)
+	if as.StringValue != "" {
+		return as.StringValue
 	}
 	return ""
 }
 
-func RemoteLogLevel(gc *types.GlobalConfig, agentName string) string {
-
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.RemoteLogLevel != "" {
-		return as.RemoteLogLevel
+func RemoteLogLevel(gc *types.ConfigItemValueMap, agentName string) string {
+	as := types.GetAgentValue("debug" + agentName + "remote.loglevel", agentName, gc)
+	if as.StringValue != "" {
+		return as.StringValue
 	}
 	return ""
 }
 
 // Ignores levels which don't parse
-func SetLogLevel(gc *types.GlobalConfig, agentName string, loglevel string) {
+func SetLogLevel(gc *types.ConfigItemValueMap, agentName string, loglevel string) {
 
 	_, err := log.ParseLevel(loglevel)
 	if err != nil {
 		log.Errorf("ParseLevel %s failed: %s\n", loglevel, err)
 		return
 	}
-	as, ok := gc.AgentSettings[agentName]
-	if ok {
-		as.LogLevel = loglevel
+	as := types.GetAgentValue("debug" + agentName + "loglevel", agentName, gc)
```

Comment:
GetAgentValue --> What is this actually doing? Is it better being a method of struct gc ??

gc.AgentSettings() --> Is this what u want? 

OR . 

gc.AgentSetting() --> Returns value of a single setting ( key )  for the given agent.